iceberg/iceberg-shading/pom.xml

```diff
@@ -0,0 +1,147 @@
+                            <execution>
+                                <phase>package</phase>
+                                <goals>
+                                    <goal>shade</goal>
+                                </goals>
+                                <configuration>
+                                    <createDependencyReducedPom>false</createDependencyReducedPom>
+                                    <relocations>
+                                        <relocation>
+                                            <pattern>org.apache.avro</pattern>
+                                            <shadedPattern>${shade.relocation.package.prefix}.org.apache.avro</shadedPattern>
+                                        </relocation>
+                                        <relocation>
+                                            <pattern>org.apache.orc</pattern>
+                                            <shadedPattern>${shade.relocation.package.prefix}.org.apache.orc</shadedPattern>
+                                        </relocation>
+                                        <relocation>
+                                            <pattern>org.apache.parquet</pattern>
+                                            <shadedPattern>${shade.relocation.package.prefix}.org.apache.parquet</shadedPattern>
+                                        </relocation>
+                                        <relocation>
+                                            <pattern>shaded.parquet</pattern>
+                                            <shadedPattern>${shade.relocation.package.prefix}.shaded.parquet</shadedPattern>
+                                        </relocation>
+                                        <relocation>
+                                            <pattern>com.google</pattern>
+                                            <shadedPattern>${shade.relocation.package.prefix}.com.google</shadedPattern>
+                                        </relocation>
+                                        <relocation>
+                                            <pattern>com.fasterxml</pattern>
+                                            <shadedPattern>${shade.relocation.package.prefix}.com.fasterxml</shadedPattern>
+                                        </relocation>
+                                        <relocation>
+                                            <pattern>com.github.benmanes</pattern>
+                                            <shadedPattern>${shade.relocation.package.prefix}.com.github.benmanes</shadedPattern>
+                                        </relocation>
+                                    </relocations>
+                                    <artifactSet>
+                                        <includes>
+                                            <include>org.apache.iceberg:*</include>
+                                            <include>org.apache.hive:hive-iceberg-catalog</include>
```

Comment:
if the catalog module depends on the shading module, isn't it a problem that the shading module's shade step depends on the catalog? Is this some form of a circular dependency? Maybe not, just checking :)

Reply:
Ah good catch that's a benign leftover, it's a noop whitelist entry in this case, as the related classes are not there yet during the build time of iceberg-shading. I'll remove it